hydra/functions/FlatteLineShape.h

```diff
@@ -78,11 +79,13 @@ namespace hydra {
  * @param daugther3_mass daughter particle 2 mass
  * @param radi decay vertex radio.
  */
-        FlatteLineShape(hydra::Parameter const& mean, hydra::Parameter const& rho1 , hydra::Parameter const& rho2,
+        FlatteLineShape(hydra::Parameter const& mean,
+                        std::vector<std::vector<double>> params,
                         double mother_mass,
                         double daugther1_mass, double daugther2_mass, double daugther3_mass,
                         double radi) :
-                BaseFunctor<FlatteLineShape<CHANNEL, ArgIndex>, hydra::complex<double>, 3>{mean, rho1, rho2},
+                BaseFunctor<FlatteLineShape<CHANNEL, ArgIndex>, hydra::complex<double>,1>{mean},
+                fParams(params),
```

Comment:
Unfortunately, this implementation does not fit the Hydra's requirements, since this class is not copyable on device side for CUDA backends. This happens because you are using a `std::vector` as a member. This will works on host and device CPU-based backends, but not in CUDA, because `std::vector` are not implemented there...